electron/src/electron.ts

```diff
@@ -73,26 +81,40 @@ if (instanceLock) {
     app.quit()
 }
 
-// Return value and exit
-ipcMain.on("cli-result", (event, result) => {
-    process.stdout.write(result);
-    process.stdout.write('\n');
-    process.exit(0)
-})
-
-// Return cli arguments
-ipcMain.on('cli-args', (event, arg) => {
-    event.returnValue = cliArgs
-})
 
-ipcMain.on('open-file-dialog', (event, options, code) => {
-    dialog.showOpenDialog(mainWindow, options).then((value) => {
-        event.sender.send('open-file-dialog-response', value, code);
-    })
-});
+const implementSyncApi = <CHANNEL extends ElectronIpcSyncRequestChannelName>(channel: CHANNEL, implementation: (...args: ElectronBridgeSyncApiRequest<CHANNEL>) => ElectronBridgeSyncApiResponse<CHANNEL>) => {
+  const responseChannelName = responseChannelNameFor(channel);
+  ipcMain.on(channel, (event, ...args) => {
+    try {
+      event.returnValue = implementation(...(args as ElectronBridgeSyncApiRequest<CHANNEL>));
+    } catch (e) {
+      event.returnValue = e;
+    }
+  })
+}
+const implementAsyncApi = <CHANNEL extends ElectronIpcAsyncRequestChannelName>(channel: CHANNEL, implementation: (...args: ElectronBridgeAsyncApiRequest<CHANNEL>) => Promise<ElectronBridgeAsyncApiResponse<CHANNEL>>) => {
+  const responseChannelName = responseChannelNameFor(channel);
+  ipcMain.on(channel, (event, code, ...args) => {
+    implementation(...args as ElectronBridgeAsyncApiRequest<CHANNEL>).then( response =>
+      event.sender.send(responseChannelName, code, response)
+    ).catch( exception =>
+      event.sender.send(responseChannelName, exceptionCodeFor(code), exception)
+    )
+  })
+}
 
-ipcMain.on('open-message-dialog', (event, options, code) => {
-    dialog.showMessageBox(options).then((value) => {
-        event.sender.send('open-message-dialog-response', value, code)
-    });
+implementSyncApi( "writeResultToStdOutAndExit", (result) => {
+  process.stdout.write(result);
+  process.stdout.write('\n');
+  process.exit(0)
 });
+implementSyncApi( "getCommandLineArguments", () => 
+  /**
+    The first two arguments are the path and the executable name, so the actual command line arguments start at index 2.
+    See: https://nodejs.org/docs/latest/api/process.html#process_process_argv.
+    The constant represents only the arguments that follow the path and executable nae.
+  */
+  process.argv.slice(2)
```

Comment:
In electron/node this is true as you run with `node script.js args` but there is a case when running a packed electron binary you don't include the script and you miss an argument eg. in linux `dicekeys args`.